independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/ConfigService.java

```diff
@@ -0,0 +1,194 @@
+package io.quarkus.analytics;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+
+import io.quarkus.analytics.config.FileLocations;
+import io.quarkus.analytics.config.GroupIdFilter;
+import io.quarkus.analytics.dto.config.AnalyticsRemoteConfig;
+import io.quarkus.analytics.dto.config.LocalConfig;
+import io.quarkus.analytics.dto.config.NoopRemoteConfig;
+import io.quarkus.analytics.dto.config.RemoteConfig;
+import io.quarkus.analytics.rest.ConfigClient;
+import io.quarkus.analytics.util.FileUtils;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+
+/**
+ * Decided the build analytics behaviour. Retrieves, stores and provides the configuration.
+ */
+public class ConfigService {
+    public static final String QUARKUS_ANALYTICS_DISABLED_LOCAL_PROP = "quarkus.analytics.disabled";
+    public static final String QUARKUS_ANALYTICS_PROMPT_TIMEOUT = "quarkus.analytics.prompt.timeout";
+    private static final int DEFAULT_REFRESH_HOURS = 12;
+    private static final String NEW_LINE = System.lineSeparator();
+    static final String ACCEPTANCE_PROMPT = NEW_LINE +
+            "*****************************************************************************************************" + NEW_LINE
+            + " Do you agree to send anonymous usage data to the Quarkus community to help improve Quarkus? (y/n) " + NEW_LINE;
```

Comment:
If the user hits 'Enter', does it default to "yes", or do something different?

Reply:
This is exactly what came to my mind when I seen `Do you agree to send anonymous usage data to the Quarkus community to help improve Quarkus? (y/n)`

`(y/n)` should be `Y/n` so when usual lazy developer comes to this, he/she hits just Enter and continues
The dev typed dev mode command and hit the `Enter` key, so his/her finder is still close to `Enter` key.

Many users just don't case about sending the data, so making it easy to agree with send the data should be a win for us.

Reply:
> And one more thing about the `quarkus.analytics.disabled` property name.

reason for this is then you can just do `export QUARKUS_ANALYTICS_DISABLED` - the mere presence disables it.

Reply:
`export QUARKUS_ANALYTICS_ENABLED=false` could work too and the property name would be in sync with the naming of the rest of Quarkus properties ...

I can live with the current naming, it's just annoyance to me that `.enabled` is not followed everywhere. There was quite an effort to unify it.

Reply:
There is a technical foundation for `QUARKUS_ANALYTICS_DISABLED`.
It's a safe default meaning it will be false all the time:
- no property -> false
- property value misspelled and set to TRRUUE-> false
It will get disable only if `QUARKUS_ANALYTICS_DISABLED=true`.

There was a month long of discussions for this precise issue on OTel.
In my opinion, consolidation around `.enabled` is pretty but also might provides a poorer solution when the default is false.